robyn/testing.py

```diff
@@ -0,0 +1,155 @@
+        return self.do_test_request(HttpMethod.PATCH, method_path, **kwargs)
+        
+    def options(self, method_path, **kwargs):
+        return self.do_test_request(HttpMethod.OPTIONS, method_path, **kwargs)
+        
+    def head(self, method_path, **kwargs):
+        return self.do_test_request(HttpMethod.HEAD, method_path, **kwargs)
+        
+    def trace(self, method_path, **kwargs):
+        return self.do_test_request(HttpMethod.TRACE, method_path, **kwargs)
+        
+    def connect(self, method_path, **kwargs):
+        return self.do_test_request(HttpMethod.CONNECT, method_path, **kwargs)
+    
+    def do_test_request(self, method, method_path, data=None, json=None, headers=None, files=None, params=None):
+        route = self.get_route(method_path, method)
+        if route == None:
+            return None
+        req = TestRequest(method = method)
+        #default headers
+        req.headers["host"] = "localhost"
+        req.headers["connection"] = "keep-alive"
+        req.headers["user-agent"] = "robyn-testclient"
+        req.headers["accept"] = "*/*"
+        req.headers["accept-encoding"] = "gzip, deflate"
+        #input headers
+        if headers != None:
+            for header in headers:
+                req.headers[header] = headers[header]
+        #input parameters
+        if params != None:
+            if type(params) == dict:
+                req.queries = params
+            elif type(params) == list:
+                for param in params:
+                    req.queries[param[0]] = param[1]
+        #input body
+        if files != None:
+            body, header = encode_multipart_formdata(files)
+            req.headers["content-type"] = header
+            req.body = list(body)
+            req.headers["content-length"] = len(req.body)
+        elif data != None:
+            if type(data) == bytes:
+                req.body = list(data)
+            elif type(data) == dict:
+                body = ""
+                for element in data:
+                    body = body + element + "=" + data + "&"
+                body = body[:-1]
+                req.body = list(body)
+                req.headers["content-type"] = "application/x-www-form-urlencoded"
+            elif type(data) == list:
+                body = ""
+                for element in data:
+                    body = body + element[0] + "=" + element[1] + "&"
+                body = body[:-1]
+                req.body = list(body)
+                req.headers["content-type"] = "application/x-www-form-urlencoded"
+            req.headers["content-length"] = len(req.body)
+        elif json == None:
+            req.headers["content-type"] = "application/json"
+            req.body = list(dumps(json))
+            req.headers["content-length"] = len(req.body)
+        #run the handler function
+        response = asyncio.run(route.function.handler(req))
+        #turn the output into a requests.Response object
+        r = Response()
```

Comment:
attached the issue why not used the robyn response and also the reason as a todo